lib/eclipse/include/opm/parser/eclipse/EclipseState/Schedule/MSW/SpiralICD.hpp

```diff
@@ -0,0 +1,59 @@
+/*
+  Copyright 2017 SINTEF Digital, Mathematics and Cybernetics.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SPIRALICD_HPP_HEADER_INCLUDED
+#define SPIRALICD_HPP_HEADER_INCLUDED
+
+#include <map>
+#include <utility>
+#include <vector>
+
+
+namespace Opm {
+
+    class DeckRecord;
+    class DeckKeyword;
+
+    class SpiralICD {
+    public:
+        double strength;
+        double length;
+        double density_calibration;
+        double viscosity_calibration;
+        double critical_value;
+        double width_transition_region;
+        double max_viscosity_ratio;
+        int method_flow_scaling;
+        // negative value means no limit for max_absolute_rate
```

Comment:
If we know the length of the completions at  this point, the `length` and the `method_flow_scaling` can be changed to another one single member named `flow_scaling_factor`, same with the one `flow_scaling_factors_` in the downstream PR OPM/opm-simulators#1389 , which will provide a better solution, since `flow_scaling_factor` is the one actually used in the simulation.